src/erlcloud_ec2.erl

```diff
@@ -1348,37 +1374,66 @@ attribute_xpath(_, AttributeName) ->
     "/DescribeInstanceAttributeResponse/" ++ AttributeName ++ "/value".
 
 
+-type ec2_instances_ids() :: [string()].
+
 %%
+%% Function for making calls to DescribeInstances action
+%% DescribeInstances Documentation: docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstances.html
 %%
--spec describe_instances() -> {ok, [proplist()]} | {error, any()}.
-describe_instances() -> describe_instances([]).
+-spec describe_instances() -> ok_error(proplist()).
+describe_instances() ->
+    describe_instances([],[],default_config()).
 
--spec describe_instances([string()] | aws_config()) -> {ok, [proplist()]} | {error, any()}.
+-spec describe_instances(aws_config()) -> ok_error(proplist());
+                        (ec2_instances_ids()) -> ok_error(proplist()).
 describe_instances(Config)
-  when is_record(Config, aws_config) ->
+    when is_record(Config, aws_config) ->
     describe_instances([], Config);
 describe_instances(InstanceIDs) ->
     describe_instances(InstanceIDs, [], default_config()).
 
--spec describe_instances([string()], filter_list() | aws_config()) -> {ok, [proplist()]} | {error, any()}.
+-spec describe_instances(ec2_instances_ids(), filter_list()) -> ok_error(proplist());
+                        (ec2_instances_ids(), aws_config()) -> ok_error(proplist());
+                        (ec2_max_result(), ec2_token()) -> ok_error(proplist(), ec2_token()).
+describe_instances(InstanceIDs, Filter)
+    when is_list(InstanceIDs), is_list(Filter) ->
+    describe_instances(InstanceIDs, Filter, default_config());
 describe_instances(InstanceIDs, Config)
-  when is_record(Config, aws_config) ->
-    describe_instances(InstanceIDs, [], Config);
-describe_instances(InstanceIDs, Filter) ->
-    describe_instances(InstanceIDs, Filter, default_config()).
+    when is_list(InstanceIDs), is_record(Config, aws_config) ->
+    describe_instances(InstanceIDs, [], Config).
 
--spec describe_instances([string()], filter_list(), aws_config()) -> {ok, [proplist()]} | {error, any()}.
+- spec describe_instances(ec2_instances_ids(), filter_list(), aws_config()) -> ok_error(proplist());
+                         (filter_list(), ec2_max_result(), ec2_token()) -> ok_error(proplist(), ec2_token()).
 describe_instances(InstanceIDs, Filter, Config)
-  when is_list(InstanceIDs) ->
+    when is_list(InstanceIDs), is_list(Filter), is_record(Config, aws_config) ->
```

Comment:
This code is very `when`-heavy. Are there any written style-guides over at AlertLogic for Erlang code? If so, it would be nice to copy/link them from this project. (My own preference in absence of any agreed style is towards picking _one_ `describe_instances/3` and users who want to filter and paginate without providing a custom Config can pass `default_config()` themselves.)